### PR TITLE
Masks to panel buffer

### LINKED ISSUE #1226: Add ability to set detector panel buffer from masks.
We could use a 3rd option for configuring the panel buffer on the detector: set from masks.  I have lost track of the logic of how the various masking options are handled @bnmajor -- but we could make it explicit regarding setting the `panel_buffer` attribute on each detector with the masks defined on that detector.

### PR DESCRIPTION
**Pull request by bnmajor**
Fixes #1226

### CODE REVIEW

File: hexrd/ui/hexrd_config.py [hunk 1]
```diff
@@ -699,8 +699,8 @@ def images_dict(self):
         return self.intensity_corrected_images_dict
 
     @property
-    def masked_images_dict(self):
-        """Get an images dict where masks have been applied"""
+    def masks_dict(self):
```
Comment (psavery) [inline hexrd/ui/hexrd_config.py:702]: Maybe we should rename this to `raw_masks_dict` to clarify that it only returns raw masks?

File: hexrd/ui/hexrd_config.py [hunk 2]
```diff
@@ -710,15 +710,29 @@ def masked_images_dict(self):
...
+            masks_dict[name] = final_mask
+
+        return masks_dict
+
+    @property
+    def masked_images_dict(self):
+        """Get an images dict where masks have been applied"""
+        images_dict = self.images_dict
+        for name, img in images_dict.items():
+            for det, mask in self.masks_dict.items():
```
Comment (psavery) [inline hexrd/ui/hexrd_config.py:733]: Can we set `masks_dict = self.masks_dict` outside of the loops so it doesn't rebuild the masks dict for every iteration of the images dict?

File: hexrd/ui/mask_manager_dialog.py [hunk 3]
```diff
@@ -306,3 +308,14 @@ def rebuild_masks(self):
             HexrdConfig().polar_masks_changed.emit()
 
         self.update_masks_list('raw')
+
+    def masks_panel_to_buffer(self):
```
Comment (psavery) [inline hexrd/ui/mask_manager_dialog.py:312]: Shouldn't this be named `masks_to_panel_buffer`?

**Comment by psavery** [comment]
I tried it locally and it seems to be working.

**Review by psavery — APPROVED** [review]
Works for me!

File: hexrd/ui/hexrd_config.py (hunk 2 above)
Comment (bnmajor) [reply hexrd/ui/hexrd_config.py]: Yes! Fixed.

File: hexrd/ui/mask_manager_dialog.py (hunk 3 above)
Comment (bnmajor) [reply hexrd/ui/mask_manager_dialog.py]: Oops, yes it should have been, thanks! Fixed the naming

**Pull request merged**